packages/promise-kit/src/promiseKit.js

```diff
@@ -42,10 +40,12 @@ export function makePromiseKit() {
   /** @type {(reason: any) => void} */
   let rej = NOOP_INITIALIZER;
 
-  const p = new HandledPromise((resolve, reject) => {
-    res = resolve;
-    rej = reject;
-  });
+  const p =
+    /** @type {Promise<T> & { domain?: any  }} */
+    (new Promise((resolve, reject) => {
```

Comment:
Are you sure? The more of the promises running around our system are handled promises, the more often we succeed at pipelining. Unless there's a reason not to, I prefer that promiseKit makes handled promises.

Reply:
Okay, duly noted.  I changed this while debugging, but I will change it back to something like:

```js
const BetterPromise = globalThis.HandledPromise || Promise;
[...]
   new BetterPromise(...)
```